docs/highlight-layer.md

```diff
@@ -0,0 +1,50 @@
+# Highlight Faux layer
+
+The highlight faux layer feature seeks to enable changing/highlighting symbol layer icon color on click. Unfortunately, Mapbox supports icon-color on sdf icons only. This introduces complexity when trying to achieve this very objective using svg icons. More details on this can be found here:
+https://github.com/mapbox/mapbox-gl-js/issues/1817
+https://stackoverflow.com/questions/38068820/how-to-change-marker-color-and-icon-on-clicking-a-marker-in-mapbox/38083323#38083323
+
+## Highlight-filter-property
```

Comment:
Nit-pick: Should we make these secondary headers lower-case to match the mapSpec property?